### DISCUSSION TITLE
VirtualBox Virtual Machine ltsp image command issue

### DISCUSSION THREAD

**Discussion by vichmartins** [ltsp/ltsp#340] [General] [answered]
Hello, this project is great! I've been using it via the chrootless method.

I've then decided it would be nicer if I used the Virtual Machine method.

I've created a VM called ltsp to serve as the server.

Then I've created a VM in VirtualBox 6.1 with the Fixed HD size and .vmdk extension, I also made sure it was a debian based install 'Ubuntu 20.04'. I called it ltsp-client-image.vmdk

I've then followed the instructions which are to create a symlink between the img to location /srv/ltsp/<IMGNAME>.img

then using the `ltsp image debian` command.
Howerver I got the following error: 
![image](https://user-images.githubusercontent.com/22303821/100942285-14627600-34c9-11eb-8b58-a489e5f122b3.png)

Is there something I am missing or did incorrectly?

PS to allow the image to be visible to the server VM I created a shared folder in VirtualBox settings to point to the image on the Host PC. Could this be the reason for it failing or is it another reason?

**Comment by alkisg** [comment]
Hi, check these answers:
https://github.com/ltsp/ltsp/discussions/295#discussioncomment-102279
https://github.com/ltsp/ltsp/discussions/285#discussioncomment-101839

**Comment by vichmartins** [comment]
Hello @alkisg thank you for your reply, and I do apoligize for the tardy response. I've gotten time to continue with the project.

After reading the 2 discussions you've sent I've remade both the ltsp server and client images to use MBR/BIOS instead of GPT. Here is an image from the client image to confirm its using MBR
![image](https://user-images.githubusercontent.com/22303821/101294969-9d8aed00-37e8-11eb-9507-9305c8f2cd14.png)

I also named the image x86_64 as recommended best practice in one of the discussion
![image](https://user-images.githubusercontent.com/22303821/101294980-b6939e00-37e8-11eb-9d46-4337285a5ff2.png)

After making these changes I tried creating the image again with `ltsp image x86_64.img`

This time I got this error:
![image](https://user-images.githubusercontent.com/22303821/101295042-1a1dcb80-37e9-11eb-9f2f-746604797f75.png)

So I looked to see if someone else has had this issue and I found discussion: #319 
However it pointed me back to #295

Which was stating it was a MBR issue which I thought I took care of but I'm problably missing I'm not seeing. Any help is greatly appreciated. Thank you for your time.

**Comment by alkisg** [comment, marked as answer]
It's late so I didn't have a good look, but I believe that now you created a separate boot partition, and LTSP is searching there for the OS.

LTSP automatically uses the first partition. To use another one, you'd need to read the advanced image sources section in https://ltsp.org/man/ltsp-ipxe/, and pass the appropriate partition parameter.

Or just reinstall and tell it to use MBR with a single ext4 /. No GPT, no separate /boot; as simple as it gets.

**Reply by vichmartins** [reply]
@alkisg Thank you for your response, I recreated everything under one partition and it worked like a charm.

I do have one more question. 

I noticed that for account/groups information changes you'd use the `ltsp initrd` command. So I created the accounts and groups settings all on the VM image. Then ran the ltsp initrd command to obtain this information. Or so I thought, however when you boot into the created image from `ltsp image x86_64.img` it'll only have the one administrator account. I noticed that it was the ltsp-server's administrator account and not the VM image's account. Is there a way I can force the initrd command obtain the VM images account and group information instead of the server?

**Comment by alkisg** [comment]
You can read about PWMERGE_SUR etc in `man ltsp.conf`.
But in that plan of yours, where does the home directory /home/username live?
Is it SSHFS? NFS? If so, where is the server, as you can't connect via SSHFS without accounts on the server.
Is it local to the clients, e.g. a partition or even tmpfs in RAM? If so, that's a valid use case.

**Reply by vichmartins** [reply]
Thank you for your reply @alkisg and again sorry for the tardy response.

My plan is to have users login in a edited desktop experience session that only allows them to use certain applications. Their home directories live in /home/$USER.
Currently I have used the default setup. so I'm using NFS no SSHFS.

I reassessed and figured it would be easier if I go the default chroot route.

I've created user accounts and edited their desktops accordingly to company rules. I net-booted from the client pc, then I logged in manually with the user's account and everything works, from the custom UI to the creating of files and edits to the user's home folder. 

I saw that I can really automate the user experience so I thought it would be great to user the AUTOLOGIN option in **ltsp.conf** so I thought to take advantage of this.

This is what I edited into the ltsp.conf

`[MAC_ADDRESS]
HOSTNAME=pc01
AUTOLOGIN=guest1
PASSWORDS_PC01="guest1/Z3Vlc3QxCg=="
`
I thought this would log me into the account I made and edited for the user. However it seemed to have logged me into what seems to be a brand new account? Also nothing would be saved on the server, as soon as I turned off the client the data written would be lost. Am I missing a setting? How would I automatically login into the user's account on the server?

**Reply by vichmartins** [reply]
@alkisg PS I  was wondering why that happen, I looked over the documentation and thought to try and add the `RELOGIN=1` parameter underneath and it seems to have worked. I don't really know why. What exactly does the `RELOGIN` option do?

**Comment by alkisg** [comment]
>  I have used the default setup. so I'm using NFS no SSHFS

The default is NFS for root /, and SSHFS for home. So the users need to have accounts on the server.

> it would be easier if I go the default chroot route

There's no default mode, but "chroot" is the hardest one. The easiest is the chrootless mode.

> Am I missing a setting?

After the "bad" login with the clean account, what is the output of the `mount` command? It should show sshfs /home.

> the RELOGIN=1 parameter

This sounds like SSHFS only worked the second time, not the first time.

**Comment by vichmartins** [comment]
@alkisg Good morning/evening. I apologize for the late response.

> There's no default mode, but "chroot" is the hardest one. The easiest is the chrootless mode.

Sorry I believe I meant to say chrootless, I think I confused the 2 there lol.

> After the "bad" login with the clean account, what is the output of the mount command? It should show sshfs /home.

So last night I ended up restarting the machines and this morning I removed the RELOGIN parameter and the normal login worked. It did not login into a temp profile like yesterday? Lol I think a service like sshfs might've been either off or stuck?
Here are the mount output the client
![client1](https://user-images.githubusercontent.com/22303821/101635575-24fa7b00-39f8-11eb-863c-fc92a91ab9a8.png)

I also saw that there exists this option: `DEFAULT_IMAGE` I thought that this might be for automatically picking an image and booting straight into it without first stopping on the ipxe menu? Is that its function? I ask because I tried applying that option onto the ltsp.conf file it still takes me to the ipxe menu. 

After  applying DEFAULT_IMAGE parameter in ltsp.conf I ran `ltsp initrd` and `ltsp ipxe -b` This still brought me to the ipxe menu.

I've tried using MENU_TIMEOUT parameters, I set it to '1' which its fast but it still shows. I tried setting it to '0' which it disables the timer all together.

Is there a way to force skip the blue ipxe menu when booting? Any help is heavily appreciated. Thank for your awesome project also I a huge fan!

**Comment by alkisg** [comment]
>  Here are the mount output the client

Yes, this has the correct sshfs mount line.

DEFAULT_IMAGE is the preselected entry in the blue menu.
MENU_TIMEOUT=-1 is supposed to hide the menu, if it doesn't, it should be fixed; check /srv/tftp/ltsp/ltsp.ipxe.

**Reply by vichmartins** [reply]
Hello @alkisg sorry for again the late response. I thank you for explaining and having patience with me. I was able to create the setup I needed to accommodate the 40 people I'm currently hosting. 

I really like this project, your work is awesome, thank you again.
